R/dist_spec.R

```diff
@@ -131,6 +131,9 @@ discrete_pmf <- function(distribution =
 #' This combines the parameters so that they can be fed as multiple delay
 #' distributions to [epinow()] or [estimate_infections()].
 #'
+#' Note that distributions that already are combinations of other distributions
```

Comment:
Related to this, can we add a more informative error for the case where one wants to nest an already nested parameter. Here, user is trying to make the mean of the mean of the Normal uncertain.

```r
> dist2 <- Gamma(
+     mean = Normal(Normal(3, 3), 0.5), sd = Normal(2, 0.5), max = 20
+ )
Error in if (rel_unc > 0) { : missing value where TRUE/FALSE needed
In addition: Warning message:
In new_dist_spec(params, "gamma", ...) :
  Uncertain gamma distribution specified in terms of parameters that are not the "natural" parameters of the distribution (shape, rate). Converting using a crude and very approximate method that is likely to produce biased results. If possible, it is preferable to specify the distribution directly in terms of the natural parameters.
```

Reply:
Fixed in 1825fe42c1d3e2021fe12aba2e2fdb9ddef9d545